File: morningSnacks/java/GreatestCommonDivisor.java
import java.util.Scanner;

public class GreatestCommonDivisor {
    public static void main(String[] args) {

        Scanner input = new Scanner(System.in);

        System.out.print("Enter the first number: ");
        int firstNumber = input.nextInt();

        System.out.print("Enter the second number: ");
        int secondNumber = input.nextInt();

        int count = 1;
        Integer largestCommonDivisor = null;

        while (count <= firstNumber / 2 || count <= secondNumber / 2) {
            if (firstNumber % count == 0 && secondNumber % count == 0) {
                largestCommonDivisor = count;
            }
            count++;
        }

        System.out.println(largestCommonDivisor);
    }
}

File: morningSnacks/python/greatestCommonDivisor.py
firstNumber = int(input("Enter the first number: "))
secondNumber = int(input("Enter the second number: "))

count = 1
largestCommonDivisor = None

while count <= firstNumber / 2 or count <= secondNumber / 2:
    if firstNumber % count == 0 and secondNumber % count == 0:
        largestCommonDivisor = count
    count += 1

print(largestCommonDivisor)

